blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScmFileContent.java

```diff
@@ -0,0 +1,139 @@
+package io.jenkins.blueocean.blueocean_github_pipeline;
+
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+/**
+ * @author Vivek Pandey
+ */
+@ExportedBean
+public class GithubScmFileContent{
```

Comment:
Right, I understand that but all the other stuff is going to be pretty much needed. It doesn't even need to be wrapped up. This would make implementing other load/save with other providers easier and more consistent, and allow the UI to just delegate to the back end what to do without having any provider specific stuff in it (e.g. gitlab doesn't have/use the sha)